packages/store/devicesSlice/selectors.js

```diff
@@ -0,0 +1,126 @@
+    return attributes.reduce(
+      (accu, value) => {
+        accu.push({ value, label: value, scope: 'identity' });
+        return accu;
+      },
+      [
+        { value: 'name', label: 'Name', scope: 'tags' },
+        { value: 'id', label: 'Device ID', scope: 'identity' }
+      ]
+    );
+  }
+);
+
+export const getDeviceCountsByStatus = createSelector([getDevicesByStatus], byStatus =>
+  Object.values(DEVICE_STATES).reduce((accu, state) => {
+    accu[state] = byStatus[state].total || 0;
+    return accu;
+  }, {})
+);
+
+export const getDeviceById = createSelector([getDevicesById, (_, deviceId) => deviceId], (devicesById, deviceId = '') => devicesById[deviceId] ?? {});
+
+export const getSelectedGroupInfo = createSelector(
+  [getAcceptedDevices, getGroupsById, getSelectedGroup],
+  ({ total: acceptedDeviceTotal }, groupsById, selectedGroup) => {
+    let groupCount = acceptedDeviceTotal;
+    let groupFilters = [];
+    if (selectedGroup && groupsById[selectedGroup]) {
+      groupCount = groupsById[selectedGroup].total;
+      groupFilters = groupsById[selectedGroup].filters || [];
+    }
+    return { groupCount, selectedGroup, groupFilters };
+  }
+);
+
+export const getLimitMaxed = createSelector([getAcceptedDevices, getDeviceLimit], ({ total: acceptedDevices = 0 }, deviceLimit) =>
+  Boolean(deviceLimit && deviceLimit <= acceptedDevices)
+);
+
+// eslint-disable-next-line no-unused-vars
+export const getGroupsByIdWithoutUngrouped = createSelector([getGroupsById], ({ [UNGROUPED_GROUP.id]: ungrouped, ...groups }) => groups);
+
+export const getGroups = createSelector([getGroupsById], groupsById => {
+  const groupNames = Object.keys(groupsById).sort();
+  const groupedGroups = Object.entries(groupsById)
+    .sort((a, b) => a[0].localeCompare(b[0]))
+    .reduce(
+      (accu, [groupname, group]) => {
+        const name = groupname === UNGROUPED_GROUP.id ? UNGROUPED_GROUP.name : groupname;
+        const groupItem = { ...group, groupId: name, name: groupname };
+        if (group.filters.length > 0) {
+          if (groupname !== UNGROUPED_GROUP.id) {
+            accu.dynamic.push(groupItem);
+          } else {
+            accu.ungrouped.push(groupItem);
+          }
+        } else {
+          accu.static.push(groupItem);
+        }
+        return accu;
+      },
+      { dynamic: [], static: [], ungrouped: [] }
+    );
+  return { groupNames, ...groupedGroups };
+});
+export const getAttributesList = createSelector(
+  [getFilteringAttributes, getFilteringAttributesFromConfig],
+  ({ identityAttributes = [], inventoryAttributes = [] }, { identity = [], inventory = [] }) =>
+    [...identityAttributes, ...inventoryAttributes, ...identity, ...inventory].filter(duplicateFilter)
+);
+
+export const getDeviceTypes = createSelector([getAcceptedDevices, getDevicesById], ({ deviceIds = [] }, devicesById) =>
+  Object.keys(
+    deviceIds.slice(0, 200).reduce((accu, item) => {
```

Comment:
so, it gets first 200 devices for type processing. enough for larger datasets?

Reply:
most likely not - it's more of a best effort for the onboarding time, but we don't have backend support for this to be usable for larger datasets 😞...